i_scene_cp77_gltf/importers/entity_import.py

```diff
@@ -661,38 +674,38 @@ def importEnt(with_materials, filepath='', appearances=[], exclude_meshes=[], in
                                     co.subtarget= bindname
 
 
+        
+        # am checking for license plates as we go rather than parsing all the components again.
+        if includes_license_plates:
+            license_plates = [obj for obj in bpy.data.objects if 'license_plate' in obj.get('componentName', '')]
+            bumper_f_objs = [obj for obj in bpy.data.objects if 'bumper_f' in obj.get('componentName', '')]
+            bumper_b_objs = [obj for obj in bpy.data.objects if 'bumper_b' in obj.get('componentName', '')]
 
-
-        #can probably a better way to pull this from somewhere but this works for now.
-        license_plates = [obj for obj in bpy.data.objects if 'license_plate' in obj.get('componentName', '')]
-        bumper_f_objs = [obj for obj in bpy.data.objects if 'bumper_f' in obj.get('componentName', '')]
-        bumper_b_objs = [obj for obj in bpy.data.objects if 'bumper_b' in obj.get('componentName', '')]
-
-        if len(license_plates) > 0:
-            for obj in license_plates:
-                #print(obj.name)
-                if not len(obj.constraints)>0:
-                    try:
-                        # use the component name to figure out if this supposed to be attached to the front or back bumper
-                        componentName = obj.get('componentName', '')
-                        bumper_type = 'bumper_f' if 'license_plate_f' in componentName else 'bumper_b'
-                        # Find the correct bumper and match it to the license plate
-                        potential_parents = bumper_f_objs if bumper_type == 'bumper_f' else bumper_b_objs
-                        # Check if there's actually an object to parent the license plate to, if there is set it to obj.parent
-                        if potential_parents:
-                            obj.parent = potential_parents[0]
-                        # I'm pretty certain you have this stored somewhere else already - we should probably look at just adding all of the localTransforms
-                        # to a dict seperate from comp earlier on and just matching them by componnent name whenever we need to apply transforms
-                            lct = next((comp for comp in comps if comp["name"]["$value"] == componentName), None)
-                            #print(lct["localTransform"])
-                            if lct:
-                                obj.location[0] = lct["localTransform"]["Position"]["x"]["Bits"]/ 131072
-                                obj.location[1] = lct["localTransform"]["Position"]["y"]["Bits"]/ 131072
-                                obj.location[2] = lct["localTransform"]["Position"]["z"]["Bits"]/ 131072
-                        else:
-                            print('no bumper found to parent license plate to')
-                    except Exception as e:
-                        print(e)
+            if len(license_plates) > 0:
+                for obj in license_plates:
```

Comment:
Might be worth putting this into an extra method for readability...?

Reply:
honestly no idea what you mean by this